docs/getstarted/upgrade-to-upbound/migrate-configurations-v2.md

```diff
@@ -0,0 +1,804 @@
+
+### 7. Connection secrets redesigned in v2
+
+:::danger
+Crossplane v2 removes built-in connection secret support from XRs entirely.
+This represents an important architectural shift in how connection details are
+managed.
+:::
+
+#### What changed
+
+**Before (v1):**
+```yaml
+apiVersion: aws.platform.upbound.io/v1alpha1
+kind: XSQLInstance
+spec:
+  parameters:
+    region: us-west-2
+  writeConnectionSecretToRef:
+    name: my-connection-secret
+    namespace: default
+```
+
+XRs had built-in `spec.writeConnectionSecretToRef` that automatically created
+a Kubernetes Secret with aggregated connection details from all composed
+resources.
+
+**After (v2):**
+```yaml
+apiVersion: aws.platform.upbound.io/v1alpha1
+kind: SQLInstance
+metadata:
+  namespace: default
+spec:
+  parameters:
+    region: us-west-2
+  # writeConnectionSecretToRef completely removed - not supported in v2 XRs
+```
+
+**The field is gone.** V2 XRs don't support automatic connection secret creation.
+
+#### The v2 philosophy
+
+According to [crossplane/crossplane#6440][xp-6440], Crossplane v2 shifts to
+representing higher-level abstractions (complete applications) rather than
+low-level infrastructure. Connection secrets are no longer a built-in feature
+because:
+
+1. Not all XRs need to expose connection details
+2. Different use cases need different secret structures
+3. Functions provide more flexibility for secret composition
+
+#### Migration path: Manually compose secrets
+
+The official guidance from the Crossplane maintainers:
+
+> "In v2 you can recreate it using functions - just have your XR compose a
+> secret with the XR connection details in it."
+
+This means your **function must explicitly create a Kubernetes Secret resource**
+containing the connection details for the XR.
+
+#### Implementation example
+
+**Step 1: Managed resources still support connection secrets**
+
+```kcl
+rdsv1beta1.Instance{
+    metadata: _metadata("rds-instance")
+    spec: {
+        forProvider: {
+            engine = "postgres"
+            username = "masteruser"
+            # ... other config
+        }
+        # Managed resources can still write connection secrets
+        writeConnectionSecretToRef = {
+            name = "{}-rds-conn".format(oxr.metadata.name)
+        }
+    }
+}
+```
+
+This creates a Secret with raw RDS credentials that the function can read via
+`ocds["rds-instance"].ConnectionDetails`.
+
+**Step 2: Function composes a user-facing Secret**
+
+```kcl
+import models.io.k8s.api.core.v1 as corev1
+
+# V2: Manually compose a Kubernetes Secret with connection details
+corev1.Secret{
+    metadata: _metadata("connection-secret") | {
+        name: "{}-connection".format(oxr.metadata.name)
+        namespace: oxr.metadata.namespace
+        labels: {
+            "crossplane.io/composite": oxr.metadata.name
+        }
+    }
+    type: "connection.crossplane.io/v1alpha1"
+    if "rds-instance" in ocds:
+        data: {
```

Comment:
when using here stringData - its much easier to read because we can leave out base64